clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/TopicSwitchWrapper.java

```diff
@@ -0,0 +1,35 @@
+package com.linkedin.davinci.kafka.consumer;
+
+import com.linkedin.venice.kafka.protocol.TopicSwitch;
+import com.linkedin.venice.pubsub.api.PubSubTopic;
+import io.tehuti.utils.Utils;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+public class TopicSwitchWrapper {
```

Comment:
Why do we need to wrap TopicSwitch?  Is this just to bundle together the message with the topic and source?

Reply:
Added this JavaDoc:

```
/**
 * This class precomputes the conversion from Avro's {@link CharSequence} to String, and also hangs on to the
 * correct instance of {@link PubSubTopic}, so that the lookup need not be performed in many locations.
 */
```